src/Utopia/Messaging/Adapter.php

```diff
@@ -71,31 +71,134 @@ protected function request(
             \curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
         }
 
-        \curl_setopt($ch, CURLOPT_URL, $url);
-        \curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
-        \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        \curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
-        \curl_setopt($ch, CURLOPT_USERAGENT, "Appwrite {$this->getName()} Message Sender");
+        \curl_setopt_array($ch, [
+            CURLOPT_CUSTOMREQUEST => $method,
+            CURLOPT_URL => $url,
+            CURLOPT_HTTPHEADER => $headers,
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_USERAGENT => "Appwrite {$this->getName()} Message Sender",
+            CURLOPT_TIMEOUT => $timeout,
+        ]);
 
         $response = \curl_exec($ch);
+
         \curl_close($ch);
 
-        if (\curl_errno($ch)) {
-            throw new \Exception('Error: '.\curl_error($ch));
+        try {
+            $response = \json_decode($response, true, flags: JSON_THROW_ON_ERROR);
+        } catch (\JsonException) {
+            // Ignore
         }
 
-        $jsonResponse = \json_decode($response, true);
+        return [
+            'url' => $url,
```

Comment:
It would be good to document why url needs to be in the response. I had to dig around a bit to figure out why.